codegen/smithy-ruby-codegen/src/main/java/software/amazon/smithy/ruby/codegen/generators/ClientGenerator.java

```diff
@@ -196,27 +192,32 @@ private void renderRbsOperations(RubyCodeWriter writer) {
     }
 
     private void renderOperation(RubyCodeWriter writer, OperationShape operation) {
-        Symbol symbol = symbolProvider.toSymbol(operation);
-        ShapeId inputShapeId = operation.getInputShape();
-        Shape inputShape = model.expectShape(inputShapeId);
+        Shape inputShape = model.expectShape(operation.getInputShape());
         Shape outputShape = model.expectShape(operation.getOutputShape());
-        String operationName =
-                RubyFormatter.toSnakeCase(symbol.getName());
-
-        String documentation = new ShapeDocumentationGenerator(model, symbolProvider, operation).render();
+        String operationName = RubyFormatter.toSnakeCase(symbolProvider.toSymbol(operation).getName());
+        boolean isStreaming = outputShape.members().stream()
+                .anyMatch((m) -> m.getMemberTrait(model, StreamingTrait.class).isPresent());
+        // optimization for client generation
+        if (isStreaming) {
+            hasStreamingOperation = true;
```

Comment:
Whats this doing? Doesn't quite seem right to set this here

Reply:
We iterate operations for streaming and if we see one, we know to generate the block io response body method. I'm also using it to have block in the method signature.

Reply:
While that does work (and I probably wrote that code originally....) - I dont think we should rely on setting a member in each of these method calls.  Would be better to set `hasStreamingOperation` once by inspecting all operations at that time.  (can still use the `isStreaming` locally, I just dont think we should be setting the instance variable here).